pkgs/servers/web-apps/pict-rs/default.nix

```diff
@@ -13,17 +13,22 @@
 
 rustPlatform.buildRustPackage rec {
   pname = "pict-rs";
-  version = "0.3.0-alpha.37";
+  version = "0.3.3";
 
   src = fetchFromGitea {
     domain = "git.asonix.dog";
     owner = "asonix";
     repo = pname;
     rev = "v${version}";
-    sha256 = "sha256-21yfsCicn2bjSNEMMWDG8wvnw10uT3M1L3cXCUhc24c=";
+    sha256 = "mEZBFDR+/aMRFw54Yq+f1gyEz8H+5IggNCpzv3UdDFg=";
   };
 
-  cargoSha256 = "sha256-F/mqdIjF5QOq5Plnq0DyeFP1+b7dCBcoU9iFxzcaZws=";
+  cargoLock = {
```

Comment:
is it wrong to use cargoLock like this in nixpkgs? 0.3 has a git dependency that was a fork of a package, and the fork was never merged. I don't want to publish a version of that fork just for pict-rs since I dropped the dependency for 0.4 anyway

Reply:
Cargolock needs to be used when there is a got dependency. So this is completely correct !

Reply:
Oh, if you use cargolock you need to commit the wholr cargo lock file to the repository, so thay nix knows where to fetch the dependencies.
If you look for other instances of cargolock usage you should see examples. Its a bit inconvenient, but its the onky way.
The good thing though is that this can be automared automatically after with nix-update

Reply:
Oh, that's interesting. the package seemed to build fine locally with the string path to `${src}/Cargo.lock`, but if it needs to be committed I can do that, also